src/test/java/com/shopify/minishopify/controllers/ShopControllerTest.java

```diff
@@ -0,0 +1,133 @@
+package com.shopify.minishopify.controllers;
+
+import com.shopify.minishopify.model.Shop;
+import com.shopify.minishopify.model.User;
+import com.shopify.minishopify.repository.ShopRepository;
+import org.json.JSONObject;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+import java.util.Optional;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ShopControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private ShopRepository shopRepository;
+
+    // test objects
+    private Shop shop1;
+    private User shopOwner;
+
+
+    @BeforeAll
+    public void initialize(){
+        shopOwner = new User("shop owner", "shopowner@email.com");
+        shop1 = new Shop(shopOwner, "Shop1", "Shop1 description");
+    }
+
+    @Test
+    public void whenFindExistingShopById_thenReturnShop() throws Exception {
+
+        when(shopRepository.findById(shop1.getId())).thenReturn(Optional.of(shop1));
+
+        mvc.perform(get("/shops/" + shop1.getId()))
+                .andDo(print())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void whenFindNonExistingShopById_thenReturnNotFound() throws Exception {
+
+        mvc.perform(get("/shops/2"))
+                .andDo(print())
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void whenUpdateExistingShop_thenReturnNewShop() throws Exception {
```

Comment:
what's difference between this and whenFindExistingShopById_thenReturnShop()

Reply:
no difference, was a mistake